src/ReadingBuddy.java

```diff
@@ -248,26 +250,122 @@ public static ArrayList<Integer> initialGetBookGroups(){
         }
     }
 
+    //NEW SPRINT3 - MOVED from User.java
+    public static ArrayList<GroupPost> getGroupPostData(ArrayList<Integer> groupNums){
+        ArrayList<GroupPost> userGroupPosts = new ArrayList<GroupPost>();
+        try { 
+            FileReader filereader = new FileReader("reading_bubby/appdata/group_posts.csv"); 
+            // create csvReader object and skip first Line 
+            CSVReader csvReader = new CSVReaderBuilder(filereader) 
+                                    .withSkipLines(1) 
+                                    .build(); 
+            List<String[]> allData = csvReader.readAll();
+            GroupPost tempPost;
+            for (String[] row : allData) {
+                tempPost = new GroupPost(row[0], row[1], Integer.parseInt(row[2]), LocalDate.parse(row[3]));
+                allGroupPosts.add(tempPost);
+                if(groupNums.contains(Integer.valueOf(row[2]))){
```

Comment:
when a GroupPost is for/from one of the user's book groups it is added to the ArrayList that gets returned (a different function sorts the user's posts into their groups)